tests/snuba/api/endpoints/test_organization_events_mep.py

```diff
@@ -3248,6 +3252,54 @@ def test_transaction_user_misery(self) -> None:
             },
         }
 
+    def test_on_demand_count_unique(self):
+        field = "count_unique(user)"
+        query = "transaction.duration:>0"
+        # We do not really have to create the metrics for both specs since
+        # the first API call will not query any on-demand metric
+        for spec_version in OnDemandMetricSpecVersioning.get_spec_versions():
+            spec = OnDemandMetricSpec(
+                field=field,
+                query=query,
+                spec_type=MetricSpecType.DYNAMIC_QUERY,
+                spec_version=spec_version,
+            )
+            self.store_on_demand_metric(1, spec=spec, timestamp=self.min_ago)
+            self.store_on_demand_metric(2, spec=spec, timestamp=self.min_ago)
+
+        # First, make a call without the next spec flag
+        response = self._make_on_demand_request({"field": [field], "query": query})
+        assert response.data == {
+            # The count is 0 because we have not made calls to self.store_event()
+            "data": [{"count_unique(user)": 0}],
+            "meta": {
+                "fields": {"count_unique(user)": "integer"},
+                "units": {"count_unique(user)": None},
+                "isMetricsData": False,
+                "isMetricsExtractedData": False,
+                "tips": {"query": None, "columns": None},
+                "datasetReason": "Column transaction.duration was not found in metrics indexer",
+                "dataset": "metricsEnhanced",
+            },
+        }
+        # Second, make a call with the next spec flag
+        # Because of the flag, it will query the 2nd spec version
+        response = self._make_on_demand_request(
+            {"field": [field], "query": query}, extra_features={NEXT_SPEC_FEAT_FLAG: True}
+        )
+        assert response.data == {
+            "data": [{"count_unique(user)": 2}],
```

Comment:
The count here is `2` rather than `0` since we made the API call with the extra feature `{NEXT_SPEC_FEAT_FLAG: True}`.